transport/src/main/java/io/netty5/channel/ChannelOutboundBuffer.java

```diff
@@ -303,7 +303,9 @@ private boolean remove0(Throwable cause, boolean notifyWritability) {
 
         if (!e.cancelled) {
             // only release message, fail and decrement if it was not canceled before.
-            safeDispose(msg);
+            if (Resource.isAccessible(msg, false)) {
```

Comment:
When we remove on failure, it may be because the resource was already closed. The check prevents double-reporting the error.
There's no issue with also adding this check to the successful remove, though.